### PR TITLE
Move Init function from parent to the struct

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: internal/data/trace_test.go [hunk 1]
```diff
@@ -52,30 +52,6 @@ func TestTraceID(t *testing.T) {
 	assert.EqualValues(t, []byte{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, tid.Bytes())
 }
 
-func TestAttrs(t *testing.T) {
```
Comment (bogdandrutu) [inline internal/data/trace_test.go:55]: Redundant with auto-generated tests.

File: internal/data/generated_common.go [hunk 2]
```diff
@@ -26,55 +26,62 @@ import (
 // This is a reference type, if passsed by value and callee modifies it the
 // caller will see the modification.
 //
-// Must use NewEmptyInstrumentationLibrary function to create new instances.
+// Must use NewInstrumentationLibrary function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type InstrumentationLibrary struct {
 	// Wrap OTLP otlpcommon.InstrumentationLibrary.
```
Comment (tigrannajaryan) [inline internal/data/generated_common.go:32]: It is worth explaining why we have a pointer-to-pointer here. We can probably say
"orig points to the pointer field contained somewhere else. We use pointer-to-pointer to be able to modify it in InitEmpty func."

File: translator/internaldata/oc_to_traces.go [hunk 3]
```diff
@@ -150,16 +150,16 @@ func ocSpanToInternal(dest data.Span, src *octrace.Span) {
 	dest.SetDroppedEventsCount(droppedEventCount)
 	dest.SetLinks(links)
 	dest.SetDroppedLinksCount(droppedLinkCount)
-	ocStatusToInternal(src.Status, dest)
+	ocStatusToInternal(dest.Status(), src.Status)
 }
 
-func ocStatusToInternal(ocStatus *octrace.Status, out data.Span) {
+func ocStatusToInternal(dest data.SpanStatus, ocStatus *octrace.Status) {
```
Comment (tigrannajaryan) [inline translator/internaldata/oc_to_traces.go:156]: This uses a different order of parameters than e.g. ocSpanToResourceSpans. Let's decide on one ordering and stick with it.

I think majority of functions use (input,output) ordering. We can also do (output,input), it does not matter, but let's make sure it is consistently used everywhere by all translation functions.

File: translator/internaldata/oc_to_traces.go [hunk 4]
```diff
@@ -150,16 +150,16 @@ func ocSpanToInternal(dest data.Span, src *octrace.Span) {
...
 	if ocStatus == nil {
 		return
 	}
-	out.InitStatusIfNil()
-	out.Status().SetCode(data.StatusCode(ocStatus.Code))
-	out.Status().SetMessage(ocStatus.Message)
+	dest.InitEmpty()
```
Comment (tigrannajaryan) [inline translator/internaldata/oc_to_traces.go:160]: Just to clarify. `InitEmpty` is a mandatory call, right? Simply having an instance of SpanStatus does not guarantee that it is initialized and `SetX` functions may be called.

Comment (bogdandrutu) [reply translator/internaldata/oc_to_traces.go:160]: Correct. Empty parents have nil embedded messages.

File: internal/data/generated_common.go (hunk 2 above)
Comment (bogdandrutu) [reply internal/data/generated_common.go:32]: Done.

File: translator/internaldata/oc_to_traces.go (hunk 4 above)
Comment (bogdandrutu) [reply translator/internaldata/oc_to_traces.go:160]: Added comment in the generated code.

File: translator/internaldata/oc_to_traces.go (hunk 3 above)
Comment (bogdandrutu) [reply translator/internaldata/oc_to_traces.go:156]: Done. went with (src, dest) pattern everywhere.

**Pull request merged**